src/java_bytecode/java_utils.cpp

```diff
@@ -20,3 +22,94 @@ bool java_is_array_type(const typet &type)
     to_struct_type(type).get_tag()),
                     "java::array[");
 }
+
+unsigned java_parameter_count(
```

Comment:
I reimplemented the two functions and refactored the code as proposed by @smowton